IBX-4962: React app block procedure updated (#1980)
ibexa/documentation-developer: docs/content_management/pages/react_app_block.md

## Create React App block

> "In the following example, you learn how to create the `Calculator` React App block [configured in the previous section's example](#react-app-block-configuration)."

Reviewer: ```suggestion
In the following example, you learn how to create the `Calculator` React App block [configured in the previous section's example](#react-app-block-configuration).
```

### Configure React App Block

> First, install the react.

Reviewer 2: ```suggestion
First, install the React.
```

Reviewer 2: I guess it should be capitalized but I may be wrong :)

Reviewer: I'm agree with this being a proper name.

```suggestion
First, install React.
```

### Configure React App Block

First, install the react.
Run `yarn add react` command.

Next, create a .jsx file which describes your component.
You can place it in any location.

> In the following example, create `Calculator.jsx` file in `/assets/page-builder/components/` directory:

Reviewer: I prefer without the leading slash.
```suggestion
In the following example, create `Calculator.jsx` file in `assets/page-builder/components/` directory:
```

### Configure React App Block

First, install the react.

> Run `yarn add react` command.

Reviewer: I don't have to do that on a 4.5.3 to make the example works.
By executing `yarn list --pattern react`, I see that it comes with `react@18.2.0`.
But it doesn't harm to ensure it's there.

As a result, the components are rendered on the page. 

``` js

> import Calculator from "/assets/page-builder/components/Calculator";

Reviewer: I prefer the relative path. Again, from my point of view, the absolute can lead to a miss interpretation of what the root is.

In vendor/ibexa, I found no use of absolute path, only relative paths or package paths. Including this insane one: https://github.com/ibexa/admin-ui/blob/main/src/bundle/Resources/public/js/scripts/widgets/flatpickr.js#L1

It was shortened in https://github.com/ezsystems/developer-documentation/pull/2074/files#diff-8f99068e452e1e02045fc9d5b04c5162e7f5ee2380a95990cf64be4742ae587aL105-R106 which is part of the conflict.

As a result, the components are rendered on the page. 

``` js

> import Calculator from "assets/page-builder/components/Calculator";

Reviewer: ```
Module build failed: Module not found:                                                                                                                
!!    "./assets/page-builder/react/blocks/Calculator.js" contains a reference to the file "assets/page-builder/components/Calculator".                      
!!    This file can not be found, please check it for typos or update it if the file got moved.
```

To fix the error above, I have to make the following change.

```suggestion
import Calculator from "/assets/page-builder/components/Calculator";
```

Before PR, we had a relative path. This change move to an absolute path.

Reviewer: Fixed in #2199

Now, you should see new `Calculator` block in the Page Builder blocks list:

![Calculator](calculator.png "Calculator - React App Block")

> Then, add new block configuration in the `templates` directory:

Reviewer: ```suggestion
Then, make sure that your [Page layout template](template_configuration.md#page-layout) (like `templates/themes/standard/pagelayout.html.twig`) has the the following Twig code in its `{% block javascripts %}`:
```

``` twig

> {% if encore_entry_exists('react-blocks-js') %}
>     { encore_entry_script_tags('react-blocks-js') }}
> {% endif %}

Reviewer: ```suggestion
{% if encore_entry_exists('react-blocks-js') %}
    {{ encore_entry_script_tags('react-blocks-js') }}
{% endif %}
```
